x/protorev/keeper/hooks.go

```diff
@@ -0,0 +1,235 @@
+		return
+	}
+
+	h.k.storeJoinExitPoolMsgs(ctx, sender, poolId, exitCoins[0].Denom, false)
+}
+
+// AfterCFMMSwap stores swaps to be checked by protorev given the coins swapped in the pool.
+func (h Hooks) AfterCFMMSwap(ctx sdk.Context, sender sdk.AccAddress, poolId uint64, input sdk.Coins, output sdk.Coins) {
+	// Checked to avoid future unintended behavior based on how the hook is called
+	if len(input) != 1 || len(output) != 1 {
+		return
+	}
+
+	h.k.storeSwap(ctx, poolId, input[0].Denom, output[0].Denom)
+}
+
+// ----------------------------------------------------------------------------
+// CONCENTRATED LIQUIDITY HOOKS
+// ----------------------------------------------------------------------------
+
+// AfterConcentratedPoolCreated checks and potentially stores the pool via the highest liquidity method.
+func (h Hooks) AfterConcentratedPoolCreated(ctx sdk.Context, sender sdk.AccAddress, poolId uint64) {
+	h.k.afterPoolCreated(ctx, poolId)
+}
+
+// AfterInitialPoolPositionCreated is a noop.
+func (h Hooks) AfterInitialPoolPositionCreated(ctx sdk.Context, sender sdk.AccAddress, poolId uint64) {
+}
+
+// AfterLastPoolPositionRemoved is a noop.
+func (h Hooks) AfterLastPoolPositionRemoved(ctx sdk.Context, sender sdk.AccAddress, poolId uint64) {
+}
+
+// AfterConcentratedPoolSwap stores swaps to be checked by protorev given the coins swapped in the pool.
+func (h Hooks) AfterConcentratedPoolSwap(ctx sdk.Context, sender sdk.AccAddress, poolId uint64, input sdk.Coins, output sdk.Coins) {
+	// Checked to avoid future unintended behavior based on how the hook is called
+	if len(input) != 1 || len(output) != 1 {
+		return
+	}
+
+	h.k.storeSwap(ctx, poolId, input[0].Denom, output[0].Denom)
+}
+
+// ----------------------------------------------------------------------------
+// HELPER METHODS
+// ----------------------------------------------------------------------------
+
+// afterPoolCreated checks if the new pool should be stored as the highest liquidity pool
+// for any of the base denoms, and stores it if so.
+func (k Keeper) afterPoolCreated(ctx sdk.Context, poolId uint64) {
+	baseDenoms, err := k.GetAllBaseDenoms(ctx)
+	if err != nil {
+		ctx.Logger().Error("Protorev error getting base denoms in AfterCFMMPoolCreated hook", err)
+		return
+	}
+
+	baseDenomMap := make(map[string]bool)
+	for _, baseDenom := range baseDenoms {
+		baseDenomMap[baseDenom.Denom] = true
+	}
+
+	pool, err := k.poolmanagerKeeper.GetPool(ctx, poolId)
+	if err != nil {
+		ctx.Logger().Error("Protorev error getting pool in AfterCFMMPoolCreated hook", err)
+		return
+	}
+
+	coins := k.getCoinsFromPool(ctx, pool, poolId)
+
+	// Pool must be active and the number of coins must be 2
+	if pool.IsActive(ctx) && len(coins) == 2 {
+		tokenA := coins[0]
+		tokenB := coins[1]
+
+		if baseDenomMap[tokenA.Denom] {
+			k.compareAndStorePool(ctx, poolId, tokenA, tokenB)
+		}
+		if baseDenomMap[tokenB.Denom] {
+			k.compareAndStorePool(ctx, poolId, tokenB, tokenA)
+		}
+	}
```

Comment:
This feels a little odd to me. We shouldn't need to make an empty coins struct just to compare denoms. It feels like we should just be directly getting the denoms from the pool here and using those in the map instead of determining this from the TotalPoolLiquidity

Reply:
The reason I grab coins instead of denoms (prior to CL was when the design decision was made) is because we compare liquidity by multiplying amounts of the coin when we call k.compareAndStorePool: https://github.com/osmosis-labs/osmosis/blob/6d7d411780b88f2da10382376061f67618ee8f1b/x/protorev/keeper/hooks.go#L185

Can switch the design to first call get denoms here, and then later call get total liquidity within compareAndStorePool, but I think I am currently leaning towards @p0mvn's suggestion: https://github.com/osmosis-labs/osmosis/pull/5045#discussion_r1212486103